sycl/source/detail/kernel_impl.cpp

```diff
@@ -57,12 +57,32 @@ kernel_impl::~kernel_impl() {
   }
 }
 
+template <info::kernel param>
+typename info::param_traits<info::kernel, param>::return_type
+get_host_kernel_info() {
+  throw runtime_error(
+      "This kernel info parameter is not supported on HOST device.");
+}
+
+template <> string_class get_host_kernel_info<info::kernel::function_name>() {
```

Comment:
this is not valid in all cases.

Reply:
See the comment for attributes parameter.